cmd/handler-api.go

```diff
@@ -288,6 +293,10 @@ func maxClients(f http.HandlerFunc) http.HandlerFunc {
 			globalHTTPStats.addRequestsInQueue(-1)
 			return
 		case <-r.Context().Done():
+			// When the client disconnects before getting the S3 handler
+			// status code response, set the status code to 499 so this request
+			// will be properly audited and traced.
+			w.WriteHeader(499)
```

Comment:
I suspect this can lead to double Writes on the responseWriter @vadmeste if ServeHTTP returned then responseWriter must have been written to already.

Reply:
> I suspect this can lead to double Writes on the responseWriter @vadmeste if ServeHTTP returned then responseWriter must have been written to already.

I thought about it, it is not possible to have a double write header. This is a handler that decided to return and not calling further other handlers in the lower layers, so it is allowed to send a status code. No handler in the upper layer is allowed to send a status code as well.

Reply:
> I suspect this can lead to double Writes

Not here, since we haven't started processing the request yet. it is canceled while in queue.